app/controllers/accept_account_invitations_controller.rb

```diff
@@ -0,0 +1,30 @@
+class AcceptAccountInvitationsController < ApplicationController
+  def show
+    not_found unless received_invitation &&
+                     invitation_belongs_to_current_user? &&
+                     not_accepted_invitation?
```

Comment:
if add the scope `unaccepted` below and use `find_by!` instead of `find_by`, no need this check.